[Foo Shi Yu] iP

CFSY opened the pull request:
### Alan
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Alan frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,


1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

CFSY committed:
Changed name to Alan and added new greeting

CFSY committed:
Completed Level-1

1. Added getFormattedReply to wrap replies in separators
2. While loop that echoes and exits on "bye"

CFSY committed:
Completed Level-2

1. Created Formatter class for formatting replies
2. Added list function

CFSY committed:
Completed Level-3

1. Added mark function
2. Created Task class

CFSY committed:
Completed Level-4

1. Added Todo, Event and Deadline that extends Task
2. Added Parser and ParsedInput to help with reading user input

CFSY committed:
Completed A-TextUiTesting

1. Added simple testcases

CFSY committed:
Completed Level-5

1. Added AlanException, InvalidValueException, NoDescriptionException, NoTimeException, NoValueException
2. Used try catch within Alan to catch AlanException

CFSY committed:
Completed Level-6

1. Added delete function
2. Added delete test cases

CFSY committed:
Commit for Enums

CFSY committed:
Refactored Week 1

Added Executor class to offload the execution of commands from Alan
Moved AlanExceptions to AlanExceptions package
Changed ParsedInput to ParsedData

CFSY committed:
Add AlanIO, FileRead, FileWrite, SaveFileException

CFSY committed:
Complete Level-7

Add FileCorruptedException
Add FileFormatter
Add FileParser

CFSY committed:
Complete Level-7

Fixed task printing bug

CFSY committed:
Included changed files missed out on prev commit

CFSY committed:
Complete Level-8

CFSY committed:
Merge branch 'branch-Level-8'

CFSY committed:
Updates Level-8

Parser now accepts time format

CFSY committed:
Complete A-MoreOOP

Rename AlanIO to Storage
Rename Formatter to Ui
Create TaskList

CFSY committed:
Complete A-Packages

Create tasks Package
Create util Package

CFSY committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

CFSY committed:
Add gradle support

CFSY committed:
Add 4 basic JUnit testcases

CFSY committed:
Build jar

CFSY committed:
Added JavaDocs for public methods

CFSY committed:
Coding standard

CFSY committed:
Fixed time formatting bug

CFSY committed:
Complete Level-9

Add NoKeywordException
Add find function

CFSY committed:
Complete Level-9

CFSY committed:
Complete A-CodingStandard

CFSY committed:
Complete A-JavaDoc

CFSY committed:
Merge branch 'branch-A-JavaDoc'

CFSY committed:
Merge branch 'branch-A-CodingStandard'

CFSY committed:
Changed date time format

CFSY committed:
Merge branch 'branch-Level-9'

zzthian commented on src/main/java/alanExceptions/AlanException.java:
```diff
@@ -0,0 +1,7 @@
+package alanExceptions;
```
Do take note, package names should not contain capital letters :)

zzthian commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,79 @@
+        return isDone ? "X" : " ";
+    }
+
+    public abstract String getTypeIcon();
```
you may add JavaDocs for public methods, even if they are abstract
😄

zzthian commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,95 @@
+     * @param index The index to retrieve from.
+     * @return The task.
+     */
+    public Task get(int index) {
```
Perhaps you can name the method to be abit clearer? Like getTask?

zzthian commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,111 @@
+ * This class is used to read and write files.
+ */
+public class Storage {
+    private Path dirPath = Paths.get("data");
```
I like how you made as many of the relevant fields private. Good coding practice. 👍

zzthian commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,78 @@
+/**
+ * This class encapsulates a Deadline Task.
+ */
+public class Deadline extends Task{
```
Do remember to put a space between the { and Task!

zzthian commented on src/main/java/alanExceptions/AlanException.java:
```diff
@@ -0,0 +1,7 @@
+package alanExceptions;
+
+public class AlanException extends Exception{
```
Don't forget to place a space between { and Exception 😬 
Found this in another part as well, might wanna look through.

zzthian reviewed (commented):
Overall great coding practices, and compliant with the coding standard required. Great job!

CFSY committed:
Fixed list bug where | is kept at the end when loadign from file

ling-ui commented on src/main/java/Alan.java:
```diff
@@ -0,0 +1,141 @@
+
+            try {
+                switch (command) {
+                    case "bye":
```
It is good that you avoided deep nesting in this case by using the switch statement 😃

ling-ui commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,119 @@
+    public ParsedData parse(InputType type, String input) throws AlanException {
+        ParsedData result;
+        switch (type) {
+
```
This line spacing can be removed

ling-ui commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,119 @@
+
+    private ParsedData parseTimedTask(String input) throws AlanException {
+        String command, task, during, time;
+        String[] first, second, third;
```
Trivial issue but can consider using clearer variable names here to increase readability of code 😃

ling-ui reviewed (commented):
Overall clear and comprehensive code, just need to fix minor coding style issues mentioned by the other reviewer. Good job 😃!

CFSY committed:
Completed Level-10

Add gui
Made executor methods return string instead of printing
Edit UI to omit separators meant for CLI

CFSY committed:
Monkey wants banana now

CFSY committed:
Complete A-Assertions

CFSY committed:
Complete A-CodeQuality

Change Variable names to be clearer
Remove redundant method calls in Parser

CFSY committed:
Merge pull request #1 from CFSY/branch-A-Assertions

Complete A-Assertions

CFSY committed:
Merge pull request #2 from CFSY/branch-A-CodeQuality

Branch a code quality

CFSY committed:
Friendlier Syntax

Currently users are unable to assign custom Keywords.

Sticking to default keywords can be cumbersome.

Letting users add custom keywords and mapping them
to the default keywords using a hashmap will allow
for efficient creation of custom keywords. Adding
a Keywords class abstracts the logic from the main
file.

CFSY committed:
Save cutom keywords to Storage

Currently custom keywords cannot be saved in
storage.

This creates annoyance for users as they will have
to re assign custom keywords everytime.

Add KeywordPair class to store a custom keyword
with the associated command. Create new read, write and
append methods in Storage to write to keywords.txt. Create
new parse methods in FileParser and format methods in FileFormatter
to translate between stored text and java objects.

Doing this will ensure future readability of code since
KeywordPair is self explanatory. Saving in a separate file reduces
formatting issues in the future.

CFSY committed:
Merge pull request #3 from CFSY/branch-C-FriendlierSyntax

Branch c friendlier syntax

CFSY committed:
Add monke personality

add banana image
add help messages
add auto resizing DialogBox

CFSY committed:
Add personality to chat bot

Add monkey themed replies
Add banana request

CFSY committed:
Shift resources from build to src

CFSY committed:
Add better GUI for monke

Add style.css for styling

CFSY committed:
Add Ui.png to docs

CFSY committed:
Update README.md

CFSY committed:
Update README.md

CFSY committed:
Update README.md

CFSY committed:
Update README.md

CFSY committed:
Update README.md

CFSY committed:
Update README.md

CFSY committed:
Renamed alan to monke

CFSY committed:
Improve codes's coding standard

CFSY committed:
Add user guide

User guide was added in a prior commit, this is for formality.
